### ISSUE TITLE
🦠 Model Request: Staphylococcus aureus Inhibitors

### ISSUE THREAD

**Issue by Richiio** [ersilia-os/ersilia#946]
### Model Name

Substructure-based search of novel antibiotics

### Model Description

The authors use a large dataset (>30k) to train an explainable graph-based model to identify potential antibiotics with low cytotoxicity. The model uses a substructure-based approach to explore the chemical space. Using this method, they were able to screen 283 compounds and identify a candidate active against methicillin-resistant S. aureus (MRSA) and vancomycin-resistant enterococci.

### Slug

antibiotics-ai

### Tag

Cytotoxicity, Antimicrobial activity, 

### Publication

https://www.nature.com/articles/s41586-023-06887-8

### Source Code

https://github.com/felixjwong/antibioticsai

### License

MIT

**Comment by GemmaTuron** [comment]
Hi @Richiio 
I've edited the description, title and slug to make them more informative. Please start by trying out the model and once you can run it in your computer I'll approve the model and create an ersilia repo.
Use this thread to explain any issues you might find in testing the model, and remember to save the python version and the version of all the packages required to run the model

**Comment by Richiio** [comment]
Thanks @GemmaTuron I'll get started. Thanks for the edits.

**Comment by Richiio** [comment]
@GemmaTuron I've ran the model on my local and it is working. The output is as seen below: 

```
(chemprop2) richiio@Richio:~/antibioticsai$ chemprop_predict --test_path working_example/test.csv --checkpoint_dir "final_checkpoints" --preds_path "working_example/test_pred.csv" --features_generator rdkit_2d_normalized --no_features_scaling
Loading training args
Setting molecule featurization parameters to default.
Loading data
283it [00:00, 278897.56it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 283/283 [00:12<00:00, 22.31it/s]Validating SMILES
Test size = 283
  0%|                                                                                                                       | 0/120 [00:00<?, ?it/s]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  1%|▉                                                                                                              | 1/120 [00:03<07:32,  3.80s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  2%|█▊                                                                                                             | 2/120 [00:08<08:28,  4.31s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  2%|██▊                                                                                                            | 3/120 [00:13<08:37,  4.42s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  3%|███▋                                                                                                           | 4/120 [00:18<08:59,  4.65s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  4%|████▋                                                                                                          | 5/120 [00:22<09:02,  4.72s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  5%|█████▌                                                                                                         | 6/120 [00:27<09:12,  4.85s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  6%|██████▍                                                                                                        | 7/120 [00:33<09:22,  4.98s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  7%|███████▍                                                                                                       | 8/120 [00:38<09:20,  5.00s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  8%|████████▎                                                                                                      | 9/120 [00:43<09:18,  5.03s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  8%|█████████▏                                                                                                    | 10/120 [00:48<09:10,  5.00s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
  9%|██████████                                                                                                    | 11/120 [00:53<09:07,  5.02s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 10%|███████████                                                                                                   | 12/120 [00:58<08:56,  4.96s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".

  0%|                                                                                                                         | 0/6 [00:00<?, ?it/s]
 17%|██████████████████▊                                                                                              |
 33%|█████████████████████████████████████▋                                                                           |
 50%|████████████████████████████████████████████████████████▌                                                        |
 67%|███████████████████████████████████████████████████████████████████████████▎                                     |
 83%|██████████████████████████████████████████████████████████████████████████████████████████████▏                  |
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████|
 11%|███████████▉                                                                                                  | 13/120 [01:03<09:06,  5.10s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 12%|████████████▊                                                                                                 | 14/120 [01:12<11:01,  6.24s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 12%|█████████████▊                                                                                                | 15/120 [01:21<12:29,  7.14s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 13%|██████████████▋                                                                                               | 16/120 [01:28<12:25,  7.17s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 14%|███████████████▌                                                                                              | 17/120 [01:36<12:24,  7.23s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 15%|████████████████▌                                                                                             | 18/120 [01:43<12:02,  7.09s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 16%|█████████████████▍                                                                                            | 19/120 [01:49<11:25,  6.78s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 17%|██████████████████▎                                                                                           | 20/120 [01:55<11:01,  6.62s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 18%|███████████████████▎                                                                                          | 21/120 [02:01<10:46,  6.53s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 18%|████████████████████▏                                                                                         | 22/120 [02:06<09:55,  6.08s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 19%|█████████████████████                                                                                         | 23/120 [02:12<09:33,  5.91s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 20%|██████████████████████                                                                                        | 24/120 [02:17<09:18,  5.82s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 21%|██████████████████████▉                                                                                       | 25/120 [02:26<10:26,  6.60s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 22%|███████████████████████▊                                                                                      | 26/120 [02:31<09:46,  6.24s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 22%|████████████████████████▊                                                                                     | 27/120 [02:37<09:20,  6.03s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 23%|█████████████████████████▋                                                                                    | 28/120 [02:42<09:06,  5.94s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 24%|██████████████████████████▌                                                                                   | 29/120 [02:48<08:44,  5.77s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 25%|███████████████████████████▌                                                                                  | 30/120 [02:54<08:38,  5.76s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 26%|████████████████████████████▍                                                                                 | 31/120 [02:59<08:23,  5.65s/it]Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".

 27%|█████████████████████████████▎                                                                                | 32/120 [03:05<08:29,  5.79s/it]
Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
 28%|██████████████████████████████▎                                                                               | 33/120 [03:10<08:11,  5.65s/it]
Loading pretrained parameter "encoder.encoder.0.cached_zero_vector".
Loading pretrained parameter "encoder.encoder.0.W_i.weight".
Loading pretrained parameter "encoder.encoder.0.W_h.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.weight".
Loading pretrained parameter "encoder.encoder.0.W_o.bias".
Loading pretrained parameter "ffn.1.weight".
Loading pretrained parameter "ffn.1.bias".
Loading pretrained parameter "ffn.4.weight".
Loading pretrained parameter "ffn.4.bias".
Loading pretrained parameter "ffn.7.weight".
Loading pretrained parameter "ffn.7.bias".
```

**Comment by Richiio** [comment]
I think we can proceed with the model incorporation

**Comment by Richiio** [comment]
Here is the output I had when using their test.csv file on the various checkpoints

[test_pred.csv](https://github.com/ersilia-os/ersilia/files/14053328/test_pred.csv)

**Comment by Richiio** [comment]
@GemmaTuron 

This model is almost complete. What is left to be done is upload all the checkpoints the authors released. The size is too large. Hence, I am uploading them in bits. Once, all models are uploaded. This model would be ready for testing.

**Comment by DhanshreeA** [comment]
Hi @Richiio I see that the PR you have created is failing because of a "No space left on device" error. There are a couple of things happening here and let me break them down one by one.

From the README of the repository, I can see that the authors mention a bunch of models that they have included within the repository:
>In the folder "final_checkpoints", there are seven directories, each with 20 ChemProp checkpoints corresponding to the final trained ensembles of models used to predict antibiotic activity against Staphylococcus aureus, cytotoxicity against HepG2 cells, cytotoxicity against human primary skeletal muscle cells, cytotoxicity against IMR-90 cells, proton motive force-altering activity in Staphylococcus aureus, antibiotic activity against Staphylococcus aureus with quinolones removed, and antibiotic activity against Staphylococcus aureus with β-lactams removed, as described in the paper. For the convenience of users, the training datasets (typically of 39,312 compounds) are reproduced from the Supplementary Datasets in the "final_checkpoints" folder.

From this it's clear that the `final_checkpoints` folder contains checkpoints for a lot more models than just antibiotic activity against Staphylococcus aureus. As a general practice, we would create different model requests for each of the different types of models mentioned here. Currently for your issue, it makes sense if you only keep checkpoints from the `final_checkpoints/antibiotic_staph` folder.

**Comment by Richiio** [comment]
Summary of literature

For antibiotics activity against Staph, we have the following:
They used a threshold of 50μM to binarize for active and inactive. This was done using an 80% normalized growth inhibition cut-off. The model they used (the saved model) is a graph neural network trained with rdkit-computed molecular features.

For human cell cytotoxicity
The training data used here was counter-screened for cytotoxicity in human liver carcinoma cells(HepG2), human primary skeletal muscle cells (HSkMCs) and human lung fibroblast cells(IMR-90). Using a concentration of 10 μM, cellular visibility was measured and their activity was binarized using a stringent 90% cell viability cut-off.

**Comment by miquelduranfrigola** [comment]
Very good, @Richiio 

I would suggest that we create 1 model for Staph only.

Then, if checkpoints size allows it, I would create a second model related to cytotoxicity including HEPG2, HSKMC and IMR-90 (so, a model with three outputs). It this is not possible, then we create one model for HEPG2, one for HSKMC and one for IMR-90. Would you agree, @DhanshreeA ?

Thanks!

**Comment by Richiio** [comment]
@DhanshreeA @miquelduranfrigola 

Please what's the way forward with this model.

**Comment by GemmaTuron** [comment]
Hi @Richiio 

As @miquelduranfrigola comments above, the idea would be to have one model with the Staphylococcus predictor and ANOTHER model with the cytotoxicity endpoints, all in one if the checkpoint size allows.

**Comment by Richiio** [comment]
@GemmaTuron 

The folders for cytotoxicity would be within the allowed bentoML space on ersilia, so I would be able to load the models. Thanks. Would create a new model issue for cytotoxicity. 

The latest pull request for this model contains just the checkpoint for antibiotic activity for staph

**Comment by GemmaTuron** [comment]
Hi @Richiio 
Please have a look at what might be causing the failure to upload to docker, check the checkpoint size and the available memory in actions:
```
[upload_model_to_dockerhub](https://github.com/ersilia-os/eos18ie/actions/runs/7787806169/job/21235775873)
The hosted runner: GitHub Actions 57 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.
```

**Comment by Richiio** [comment]
@GemmaTuron @HellenNamulinda 

I have not the slightest idea why it is not building on docker. This could be a storage issue or a network issue because I proceeded to test on codespaces and got the required response as seen below:
```
🚀 Serving model eos18ie: antibiotics-ai

   URL: http://127.0.0.1:34815
   PID: 16541
   SRV: conda

👉 To run model:
   - run

💁 Information:
   - info
⛔ Model eos18ie closed
🚀 Serving model eos18ie: antibiotics-ai

   URL: http://127.0.0.1:58603
   PID: 17386
   SRV: conda

👉 To run model:
   - run

💁 Information:
   - info
(base) root@codespaces-79e63a:/workspaces/ersilia/eos18ie# ersilia run -i "C1=C(SC(=N1)SC2=NN=C(S2)N)[N+](=O)[O-]"
{
    "input": {
        "key": "NQQBNZBOOHHVQP-UHFFFAOYSA-N",
        "input": "C1=C(SC(=N1)SC2=NN=C(S2)N)[N+](=O)[O-]",
        "text": "C1=C(SC(=N1)SC2=NN=C(S2)N)[N+](=O)[O-]"
    },
    "output": {
        "outcome": [
            0.1602096880087629
        ]
    }
}
(base) root@codespaces-79e63a:/workspaces/ersilia/eos18ie# 
```

We could however reduce the number of checkpoints to 15 as opposed to the 20 used by the authors. If we find out that storage is the issue then, the other model would have to be split into 3 as opposed to the single model we wanted. I'm open to other suggestions on this matter @miquelduranfrigola

**Comment by miquelduranfrigola** [comment]
Thanks @Richiio 
Have we tried to re-run the GitHub Actions workflow?
In GitHub Codespaces, could you check how much memory is being used by the model?
GitHub Actions runners are probably smaller than Codespaces machines (just guessing), so perhaps this can be the reason?

**Comment by Richiio** [comment]
@miquelduranfrigola @GemmaTuron 

On codespaces, I have a storage usage of 15.28GB

**Comment by GemmaTuron** [comment]
what is the limit for GitHub Actions?

**Comment by Richiio** [comment]
@GemmaTuron How do I check that?

**Comment by GemmaTuron** [comment]
@Richiio this is a simple google search: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners

**Comment by GemmaTuron** [comment]
indeed @miquelduranfrigola the limit for codespaces is higher, with 32 GB of memory: https://docs.github.com/en/codespaces/overview

**Comment by GemmaTuron** [comment]
We can potentially reduce the number of checkpoints but to do that we must ensure we are not losing information. 
@Richiio let's compare their performance - ideally, with a dataset that is annotated for S. aureus activity (maybe from ChEMBL?) but otherwise at least compare the values obtained on a random dataset of molecules with 10 vs 20 checkpoints, see if that changes significantly the results
With that, we can maybe safely reduce the number of checkpoints and therefore model size

**Comment by Richiio** [comment]
@GemmaTuron 

I'll work on that. I'll make a comparison with the two datasets. One with EML and the other with new data from ChemBL. I'll compare them using just 10rows.

**Comment by GemmaTuron** [comment]
Hi @Richiio 

I think the best would be to use a large dataset, from minimum 100 rows, so that we can build a scatter plot, which will visually show better the correlation between the smaller and the larger model, what do you think?
When you find a good ChEMBL dataset indicate the link here so we can see where the data is coming from, thanks!

**Comment by HellenNamulinda** [comment]
@GemmaTuron and @Richiio,
That error seems not conclusive, [Build and push](https://github.com/ersilia-os/eos18ie/actions/runs/7787806169/job/21235775873#step:6:22and) the logs are not avaialable.

Can try to re-run. 
Because the github hosted runners(public repositories) seem to actually have more space(16GB RAM and 150GB  storage) than code spaces.

The Model size isn't that big. Checkpoints alone consume lessthan 1GB(881.4MB), and the compiled Mode size is [1692.757246017456 MB](https://github.com/ersilia-os/eos18ie/actions/runs/7786930454/job/21232868813#step:9:1682), unless it requires more RAM like some models we worked on before.

But it could have been a network outage and i request we first re-run so we can get better logs.

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda 

I thought Sarima already tried the rerunning, I'll set this now

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda and @Richiio 
The action still fails, we might indeed be hitting the 16 GB RAM limit - codespaces is now fetching the model from GitHub, which requires less space in general than docker, so that might be why Codespaces is working but we cannot upload the model to docker.
@miquelduranfrigola we never got to set a manual pipeline for uploading to docker, right?

**Comment by HellenNamulinda** [comment]
Indeed, docker might be requiring more RAM.
@Richiio Let's examine this in more detail.

**Comment by Richiio** [comment]
@GemmaTuron @HellenNamulinda 

I was able to get data from Chembl of Staph inhibitors. The data can be found here 
[Staphylococcus_aureus_assays.zip](https://github.com/ersilia-os/ersilia/files/14183733/Staphylococcus_aureus_assays.zip)
 This data was preprocessed and I got the following file 
[Chembl.csv](https://github.com/ersilia-os/ersilia/files/14183740/Chembl.csv)

The above data contained 138086 rows of data which was took too much time to load and produce result. Hence, I reduced them to 150 rows of data which is this dataset
[chembl_data.csv](https://github.com/ersilia-os/ersilia/files/14183772/chembl_data.csv)

I ran predictions using different number of checkpoints, as in 20, 15, first 10 and last10. The output files are shown below:
All the 20 checkpoints
[output_chem.csv](https://github.com/ersilia-os/ersilia/files/14183795/output_chem.csv)

First 15 checkpoint
[output_chem15.csv](https://github.com/ersilia-os/ersilia/files/14183808/output_chem15.csv)

First 10 checkpoint
[output_chem10.csv](https://github.com/ersilia-os/ersilia/files/14183810/output_chem10.csv)

Last 10 checkpoint
[output_chem11.csv](https://github.com/ersilia-os/ersilia/files/14183813/output_chem11.csv)

Using 15 checkpoints was closest to the results gotten when I used all the checkpoints

**Comment by Richiio** [comment]
@HellenNamulinda @GemmaTuron 
For a better visual view, we have this table showing the first 10 compounds and their respective results for the various checkpoints

<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

canonical smiles | 20 checkpoints | 15 checkpoints | First 10 checkpoints | Last 10 checkpoints
-- | -- | -- | -- | --
 C=C1C[C@@]23C=CC(=O)[C@@](C)(CCC(=O)Nc4c(O)ccc(C(=O)O)c4O)[C@@H]2C[C@@H]1CC3 | 0.1094783078879118 | 0.10764369318882624 | 0.1164815790951252 | 0.10247503668069839


</body>
</html>

**Comment by GemmaTuron** [comment]
Thanks @Richiio !

Let's also build the scatter plots to see better the performance between models (compare each reduced fold with the baseline 20 folds). Once this is done, we could try to incorporate the 15 checkpoints.

**Comment by GemmaTuron** [comment]
In parallel to building the scatter plots, let's upload the model with 1 checkpoint only to see if this is the cause of the memory issue, if it is a memory issue inded

**Comment by GemmaTuron** [comment]
@Richiio 

Please add all the versions of the package

**Comment by Richiio** [comment]
@GemmaTuron @HellenNamulinda 
Ersilia is currently failing to install packages. Using the current dockerfile but specifying its versions. When testing on codespace using repo-path, I obtained the following error:
```
[]
🚨🚨🚨 Something went wrong with Ersilia 🚨🚨🚨

Error message:

Ersilia exception class:
EmptyOutputError

Detailed error:
Model API eos18ie:run did not produce an outputFailed to find the pandas get_adjustment() function to patch
Failed to patch pandas - PandasTools will have limited functionality
Traceback (most recent call last):
  File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/main.py", line 6, in <module>
    import chemprop
  File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/chemprop/__init__.py", line 10, in <module>
    import chemprop.hyperparameter_optimization
  File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/chemprop/hyperparameter_optimization.py", line 8, in <module>
    from hyperopt import fmin, tpe, Trials
ModuleNotFoundError: No module named 'hyperopt'
```
Here is my dockerfile
```
FROM bentoml/model-server:0.11.0-py310
MAINTAINER ersilia

RUN pip install rdkit==2023.9.4
RUN pip install chemprop
RUN pip install descriptastorus==2.5.0.23
RUN pip install tqdm==4.66.1
RUN pip install typed-argument-parser==1.6.1
RUN pip install scikit-learn==1.4.0
RUN pip install torch==2.2.0
RUN pip install pandas==2.2.0
RUN pip install tensorboardX==2.0
RUN pip install hyperopt==0.2.7

WORKDIR /repo
COPY . /repo
```

This is similar to the error the second model produced. 
This showed that the module hyperopt wasn't found, which was explicitly written in the dockerfile. I had the same error with tensorboardX but since, this wasn't needed in our model, I commented out the import statement from chemprop and the model passed this stage. However, the code broke when it encountered other package

**Comment by GemmaTuron** [comment]
Hi @Richiio 
Have you tried to build the environment manually and then run the model? Does it fail there? 
Also please, manually activate the environment created by Ersilia and check that hyperopt is actually installed

**Comment by Richiio** [comment]
@GemmaTuron @HellenNamulinda 
The list of packages in the eos18ie environment created by Ersilia is shown below:
```
(eos18ie) root@codespaces-79e63a:/workspaces/ersilia/eos18ie# conda list
# packages in environment at /root/miniconda3/envs/eos18ie:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
alembic                   1.13.1                   pypi_0    pypi
bentoml                   0.11.0                   pypi_0    pypi
blinker                   1.7.0                    pypi_0    pypi
boto3                     1.34.36                  pypi_0    pypi
botocore                  1.34.36                  pypi_0    pypi
bzip2                     1.0.8                h7b6447c_0  
ca-certificates           2023.12.12           h06a4308_0  
cerberus                  1.3.5                    pypi_0    pypi
certifi                   2024.2.2                 pypi_0    pypi
chardet                   5.2.0                    pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
cloudpickle               3.0.0                    pypi_0    pypi
descriptastorus           2.5.0.23                 pypi_0    pypi
docker                    7.0.0                    pypi_0    pypi
filelock                  3.13.1                   pypi_0    pypi
flask                     3.0.2                    pypi_0    pypi
fsspec                    2024.2.0                 pypi_0    pypi
future                    0.18.3                   pypi_0    pypi
greenlet                  3.0.3                    pypi_0    pypi
humanfriendly             10.0                     pypi_0    pypi
hyperopt                  0.2.7                    pypi_0    pypi
idna                      3.6                      pypi_0    pypi
itsdangerous              2.1.2                    pypi_0    pypi
jinja2                    3.1.3                    pypi_0    pypi
jmespath                  1.0.1                    pypi_0    pypi
joblib                    1.3.2                    pypi_0    pypi
ld_impl_linux-64          2.38                 h1181459_1  
libffi                    3.4.4                h6a678d5_0  
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libstdcxx-ng              11.2.0               h1234567_1  
libuuid                   1.41.5               h5eee18b_0  
mako                      1.3.2                    pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
multidict                 6.0.5                    pypi_0    pypi
mypy-extensions           1.0.0                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0  
networkx                  3.2.1                    pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
nvidia-cublas-cu12        12.1.3.1                 pypi_0    pypi
nvidia-cuda-cupti-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-nvrtc-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-runtime-cu12  12.1.105                 pypi_0    pypi
nvidia-cudnn-cu12         8.9.2.26                 pypi_0    pypi
nvidia-cufft-cu12         11.0.2.54                pypi_0    pypi
nvidia-curand-cu12        10.3.2.106               pypi_0    pypi
nvidia-cusolver-cu12      11.4.5.107               pypi_0    pypi
nvidia-cusparse-cu12      12.1.0.106               pypi_0    pypi
nvidia-nccl-cu12          2.19.3                   pypi_0    pypi
nvidia-nvjitlink-cu12     12.3.101                 pypi_0    pypi
nvidia-nvtx-cu12          12.1.105                 pypi_0    pypi
openssl                   3.0.13               h7f8727e_0  
packaging                 23.2                     pypi_0    pypi
pandas                    2.2.0                    pypi_0    pypi
pandas-flavor             0.6.0                    pypi_0    pypi
pillow                    10.2.0                   pypi_0    pypi
pip                       23.3.1          py310h06a4308_0  
prometheus-client         0.19.0                   pypi_0    pypi
protobuf                  3.18.3                   pypi_0    pypi
psutil                    5.9.8                    pypi_0    pypi
py4j                      0.10.9.7                 pypi_0    pypi
python                    3.10.13              h955ad1f_0  
python-dateutil           2.8.2                    pypi_0    pypi
python-json-logger        2.0.7                    pypi_0    pypi
pytz                      2024.1                   pypi_0    pypi
rdkit                     2023.9.4                 pypi_0    pypi
readline                  8.2                  h5eee18b_0  
requests                  2.31.0                   pypi_0    pypi
ruamel-yaml               0.18.6                   pypi_0    pypi
ruamel-yaml-clib          0.2.8                    pypi_0    pypi
s3transfer                0.10.0                   pypi_0    pypi
scikit-learn              1.4.0                    pypi_0    pypi
scipy                     1.12.0                   pypi_0    pypi
setuptools                68.2.2          py310h06a4308_0  
six                       1.16.0                   pypi_0    pypi
sqlalchemy                2.0.25                   pypi_0    pypi
sqlalchemy-utils          0.41.1                   pypi_0    pypi
sqlite                    3.41.2               h5eee18b_0  
sympy                     1.12                     pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tensorboardx              2.0                      pypi_0    pypi
threadpoolctl             3.2.0                    pypi_0    pypi
tk                        8.6.12               h1ccaba5_0  
torch                     2.2.0                    pypi_0    pypi
tqdm                      4.66.1                   pypi_0    pypi
triton                    2.2.0                    pypi_0    pypi
typed-argument-parser     1.6.1                    pypi_0    pypi
typing-extensions         4.9.0                    pypi_0    pypi
typing-inspect            0.9.0                    pypi_0    pypi
tzdata                    2023.4                   pypi_0    pypi
urllib3                   2.0.7                    pypi_0    pypi
werkzeug                  3.0.1                    pypi_0    pypi
wheel                     0.41.2          py310h06a4308_0  
xarray                    2024.1.1                 pypi_0    pypi
xz                        5.4.5                h5eee18b_0  
zlib                      1.2.13               h5eee18b_0  
```

**Comment by HellenNamulinda** [comment]
> @GemmaTuron @HellenNamulinda Ersilia is currently failing to install packages. Using the current dockerfile but specifying its versions. When testing on codespace using repo-path, I obtained the following error:
> 
> ```
> []
> 🚨🚨🚨 Something went wrong with Ersilia 🚨🚨🚨
> 
> Error message:
> 
> Ersilia exception class:
> EmptyOutputError
> 
> Detailed error:
> Model API eos18ie:run did not produce an outputFailed to find the pandas get_adjustment() function to patch
> Failed to patch pandas - PandasTools will have limited functionality
> Traceback (most recent call last):
>   File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/main.py", line 6, in <module>
>     import chemprop
>   File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/chemprop/__init__.py", line 10, in <module>
>     import chemprop.hyperparameter_optimization
>   File "/root/eos/repository/eos18ie/20240207110603_D3B09D/eos18ie/artifacts/framework/code/chemprop/hyperparameter_optimization.py", line 8, in <module>
>     from hyperopt import fmin, tpe, Trials
> ModuleNotFoundError: No module named 'hyperopt'
> ```
> 
> Here is my dockerfile
> 
> ```
> FROM bentoml/model-server:0.11.0-py310
> MAINTAINER ersilia
> 
> RUN pip install rdkit==2023.9.4
> RUN pip install descriptastorus==2.5.0.23
> RUN pip install tqdm==4.66.1
> RUN pip install typed-argument-parser==1.6.1
> RUN pip install scikit-learn==1.4.0
> RUN pip install torch==2.2.0
> RUN pip install pandas==2.2.0
> RUN pip install tensorboardX==2.0
> RUN pip install hyperopt==0.2.7
> 
> WORKDIR /repo
> COPY . /repo
> ```
> 
> This is similar to the error the second model produced. This showed that the module hyperopt wasn't found, which was explicitly written in the dockerfile. I had the same error with tensorboardX but since, this wasn't needed in our model, I commented out the import statement from chemprop and the model passed this stage. However, the code broke when it encountered other package

@Richiio,
I think you meant the GitHub action for [Model Test on PR](https://github.com/ersilia-os/eos18ie/actions/runs/7813121058) is failing? Right?

Previously, we had specifed the version for tensorboard x to 2.0, also as seen in that conda enviromnt you shared(but instead of 2.0, you're installing 2.6.2.2) , and there is an a conflict for that as seen in the log for the [GitHub Action](https://github.com/ersilia-os/eos18ie/actions/runs/7813121058/job/21311605138?pr=3#step:9:1533)

**Comment by Richiio** [comment]
@GemmaTuron 

The test on PR has passed and is ready to be merged. The scatter plot showing the results is as seen below
![scatter](https://github.com/ersilia-os/ersilia/assets/49585183/99052451-e40b-4583-ad9c-8d3766859911)

**Comment by Richiio** [comment]
@GemmaTuron 
There's only one checkpoint in the PR now

**Comment by HellenNamulinda** [comment]
@GemmaTuron and @miquelduranfrigola, 
Have we been able to upload any model to docker hub using github actions in the last week?. 
Because this same model that [failed to build in ersilia's actions](https://github.com/ersilia-os/eos18ie/actions/runs/7829657562)(because of time) actually passed with a[ github action on my fork.](https://github.com/HellenNamulinda/eos18ie/actions/runs/7842536710/job/21401065646)

Was the network stable while running the action? Because it took over six hours.
Could there be a issue with ersilia's action or ersilia org?

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda 
I was just looking into it, in the same line of thought as you. Let me get back on this

**Comment by miquelduranfrigola** [comment]
Hello @HellenNamulinda - I see. It seems to be a general problem as you suggest. I have triggered workflows for the simplest and lightest model: https://github.com/ersilia-os/eos3b5e/ - let's see if they work. If DockerHub does not work in this case, then we'll troubleshoot. I see many deprecation warnings in the "upload to dockerhub" action, perhaps it is related to that.

On a quick look, I think credentials in DockerHub are OK and also I don't think we've reached any limit in the public DockerHub ersiliaio profile.

**Comment by GemmaTuron** [comment]
Ok, I was going to do that test, thanks @miquelduranfrigola.
The build has worked for the eos3b5e model, so there is nothing in the pipeline. I think chemprop itself, due to dependencies and so might be the problem? I am not familiar with local docker build, is it using the exact same structure and including the Ersilia image and all, which we know is quite heavy? Maybe the local is just a test?

BTW, if we use @emmakodes file for local testing, we might want to add it to Ersilia's repo instead of fetching from his repo? https://raw.githubusercontent.com/emmakodes/checkbuild/main/local_place_a_dockerfile_in_current_eos_repo.py

**Comment by HellenNamulinda** [comment]
At some point, I'm forced to think we are getting random errors :face_with_thermometer: 

Yes, we're using @emmakodes file for local testing,

But after some time with @Richiio, this is my suggestion. 
We update the action for uploading to docker. We add a job for cleaning up the disk space before build.
This comes after running the workflow with all the 20 checkpoints and also 15,[ complained of space.](https://github.com/HellenNamulinda/eos18ie/actions/runs/7844062384)

And inspired by [this discussion](https://github.com/orgs/community/discussions/26351),  Adding a job to [clean up the disk](https://github.com/HellenNamulinda/eos18ie/actions/runs/7844945005/workflow#L28) before building the image helped. 
[Docker build was successful](https://github.com/HellenNamulinda/eos18ie/actions/runs/7844945005/job/21408299511) with all the 20 checkpoints uploaded.

**Comment by miquelduranfrigola** [comment]
OK, fantastic stuff, @HellenNamulinda 

Happy to include the disk clean up step. This actually looks very good.

If you haven't yet, I suggest we do a PR on the [ersilia-os/eos18ie](https://github.com/ersilia-os/eos18ie) repo containing this extra step in the workflow. If it works, we'll replicate it to other models and, most importantly, in the [ersilia-os/eos-template](https://github.com/ersilia-os/eos-template) repository.

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda , brilliant! thanks
@Richiio make sure to add this step in the Cytotox models as well

**Comment by Richiio** [comment]
@GemmaTuron @miquelduranfrigola @HellenNamulinda 

The pull request I made had only one checkpoint. I'm creating a new PR for one with 15 checkpoints.

**Comment by GemmaTuron** [comment]
Hi @Richiio 

Please make a PR with the 20 checkpoints as Hellen indicated, this works on her local.

**Comment by GemmaTuron** [comment]
This model is completed with the 20 checkpoints, we can close this model request!
